configure.ac

```diff
@@ -2532,6 +2532,7 @@ else
     EXPAND_VARIABLE(sysconfdir, e_sysconfrulesdir, "/suricata/rules")
     EXPAND_VARIABLE(localstatedir, e_localstatedir, "/run/suricata")
     EXPAND_VARIABLE(datadir, e_datarulesdir, "/suricata/rules")
+    EXPAND_VARIABLE(localstatedir, e_datadir, "/lib/suricata/data")
```

Comment:
This directory appears to have 2 usages.  One as a place to hold datasets that may be distributed as part of a ruleset, but the other is more like a cache, for example the state files.

I like to separate the 2, so I can `rm -rf` the cache type files to reset my state, but not lose files that may be required to startup.

It would also be nice to set some guidelines for people who wish to distribute rules that make use of datasets.